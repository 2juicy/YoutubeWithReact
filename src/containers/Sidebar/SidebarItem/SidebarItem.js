import React from "react";
import { Menu, Icon } from "semantic-ui-react";
import "./SidebarItem.scss";

export function SidebarItem({ highlight, icon, label }) {
  return (
    <Menu.Item className={`sidebar-item ${highlight ? "highlight" : null}`}>
      <div className="sidebar-item-alignment">
        <span>
          <Icon size="large" name={icon} />
        </span>
        <span>{label}</span>
      </div>
    </Menu.Item>
  );
}
