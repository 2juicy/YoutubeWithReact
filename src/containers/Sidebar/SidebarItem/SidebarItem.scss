.sidebar-item {
  background-color: #f5f5f5;
  max-width: 17rem;
  span {
    i.icon {
      margin-right: 20px;
      color: gray;
    }
  }

  &.highlight {
    span {
      font-weight: bold;
    }
    i.icon {
      color: red;
    }
  }
}

.sidebar-item-align {
  display: flex;
  align-items: center;
}
