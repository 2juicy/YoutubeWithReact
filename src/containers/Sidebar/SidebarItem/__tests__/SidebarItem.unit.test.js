import React from "react";
import { shallow } from "enzyme";
import SidebarItem from "../SidebarItem";

describe("SidebarItem", () => {
  test("renders empty SidebarItem", () => {
    const wrapper = shallow(<SidebarItem />);
    expect(wrapper).toMatchSnapshot();
  });
});

test("renders highlighted SideBarItem that navigates to /feed/trending", () => {
  const wrapper = shallow(<SidebarItem label="Trending" icon="fire" />);
  expect(wrapper).toMatchSnapshot();
});
