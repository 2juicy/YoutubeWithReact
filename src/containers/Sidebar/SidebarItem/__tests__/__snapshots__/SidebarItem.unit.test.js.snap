// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem renders empty SidebarItem 1`] = `
<MenuItem
  className="sidebar-item null"
>
  <div
    className="sidebar-item-align"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`renders highlighted SideBarItem that navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item null"
>
  <div
    className="sidebar-item-align"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
